libsolidity/analysis/ViewPureChecker.h

```diff
@@ -75,6 +75,7 @@ class ViewPureChecker: private ASTConstVisitor
 	langutil::ErrorReporter& m_errorReporter;
 
 	bool m_errors = false;
+	bool m_checkModifiers = false;
```

Comment:
I haven't thought this through entirely - but would it be possible to only visit functions and modifiers once, but to visit the referenced modifier when encountering the modifier invocation on a function unless ``m_inferredMutability`` is already populated for that modifier?
To me that would seem cleaner than this two-step approach.

Reply:
it seems to work!